src/audio/volume/volume.c

```diff
@@ -344,6 +345,19 @@ static struct comp_dev *volume_new(const struct comp_driver *drv,
 	if (!cd)
 		goto fail;
 
+	/* malloc memory to store current volume 4 times to ensure the address
+	 * is 8-byte aligned for multi-way xtensa intrinsic operations.
+	 */
+	cd->vol = rzalloc(SOF_MEM_ZONE_RUNTIME, 0, SOF_MEM_CAPS_RAM, vol_size);
+	if (!cd->vol)
+		goto cd_fail;
+
+	address = (int32_t)cd->vol;
+	if (address % 8 != 0) {
```

Comment:
this is impossible. your math above is 4 * size of uint32t (aka 4).

so n * 4 * 4 = 16n. 16n / 8  will always divide without a remainder into 2n. So this check is not needed.

Reply:
nvm I got my check vs address math mixed up